tkg/client/upgrade_cluster.go

```diff
@@ -1184,18 +1214,49 @@ func (c *TkgClient) handleKappControllerUpgrade(regionalClusterClient, currentCl
 		return errors.Wrapf(err, "unable to delete existing kapp-controller")
 	}
 
-	// Update AWSCluster cniIngressRules to include kapp-controller API port only if CAPA is running on the management cluster
+	// handle aws management cluster
 	if err := regionalClusterClient.GetResource(&corev1.Namespace{}, clusterclient.CAPAControllerNamespace, clusterclient.CAPAControllerNamespace, nil, nil); err != nil {
-		// if capa-system namespace doesn't exist, then assume that updates to AWSCluster are not required during upgrade.
 		if !apierrors.IsNotFound(err) {
 			return errors.Wrapf(err, "unable to check if Cluster API Provider for AWS is enabled")
 		}
 	} else {
-		if err := regionalClusterClient.UpdateAWSCNIIngressRules(upgradeClusterConfig.ClusterName, upgradeClusterConfig.ClusterNamespace); err != nil {
-			return errors.Wrapf(err, "unable to update AWS CNI ingress rules")
+		kappControllerAWSIngressRule := capav1beta2.CNIIngressRule{
+			Description: "kapp-controller",
+			Protocol:    capav1beta2.SecurityGroupProtocolTCP,
+			FromPort:    DefaultKappControllerHostPort,
+			ToPort:      DefaultKappControllerHostPort,
+		}
+		err := addAWSIngressrule(regionalClusterClient, upgradeClusterConfig, kappControllerAWSIngressRule)
+		if err != nil {
+			return err
 		}
 	}
+	return nil
+}
 
+// PrepareAddonsManagerUpgrade is only meant for testing prepareAddonsManagerUpgrade
+func (c *TkgClient) PrepareAddonsManagerUpgrade(regionalClusterClient clusterclient.Client, upgradeClusterConfig *ClusterUpgradeInfo) error {
+	return c.prepareAddonsManagerUpgrade(regionalClusterClient, upgradeClusterConfig)
+}
+
+func (c *TkgClient) prepareAddonsManagerUpgrade(regionalClusterClient clusterclient.Client, upgradeClusterConfig *ClusterUpgradeInfo) error {
+	// handle aws management cluster
+	if err := regionalClusterClient.GetResource(&corev1.Namespace{}, clusterclient.CAPAControllerNamespace, clusterclient.CAPAControllerNamespace, nil, nil); err != nil {
```

Comment:
This seems to be checked again in addAWSIngressrule. Do we need this check here?

Reply:
I think, maybe we go the other way and take it out of addAWSIngressrule. since the name "addAWSingressrule" seems to require that the cluster be an AWS cluster :)

Reply:
Removed the checks from addAWSIngress